homeassistant/components/overkiz/climate_entities/atlantic_pass_apc_zone_control_zone.py

```diff
@@ -0,0 +1,252 @@
+    """Representation of Atlantic Pass APC Heating And Cooling Zone Control."""
+
+    def __init__(
+        self, device_url: str, coordinator: OverkizDataUpdateCoordinator
+    ) -> None:
+        """Init method."""
+        super().__init__(device_url, coordinator)
+
+        # There is less supported functions, because they depend on the ZoneControl.
+        if not self.is_using_derogated_temperature_fallback:
+            # Modes are not configurable, they will follow current HVAC Mode of Zone Control.
+            self._attr_hvac_modes = []
+
+            # Those are available and tested presets on Shogun.
+            self._attr_preset_modes = [*PRESET_MODES_TO_OVERKIZ]
+
+        # Those APC Heating and Cooling probes depends on the zone control device (main probe).
+        # Only the base device (#1) can be used to get/set some states.
+        # Like to retrieve and set the current operating mode (heating, cooling, drying, off).
+        self.zone_control_device = self.executor.linked_device(
+            TEMPERATURE_ZONECONTROL_DEVICE_INDEX
+        )
+
+    @property
+    def is_using_derogated_temperature_fallback(self) -> bool:
+        """Check if the device behave like the Pass APC Heating Zone."""
+
+        return self.executor.has_command(
+            OverkizCommand.SET_DEROGATED_TARGET_TEMPERATURE
+        )
+
+    @property
+    def zone_control_hvac_mode(self) -> HVACMode:
+        """Return hvac operation ie. heat, cool, dry, off mode."""
+
+        if (
+            state := self.zone_control_device.states[
+                OverkizState.IO_PASS_APC_OPERATING_MODE
+            ]
+        ) is not None and (value := state.value_as_str) is not None:
+            return OVERKIZ_TO_HVAC_MODE[value]
+        return HVACMode.OFF
+
+    @property
+    def hvac_mode(self) -> HVACMode:
+        """Return hvac operation ie. heat, cool, dry, off mode."""
+
+        if self.is_using_derogated_temperature_fallback:
+            return super().hvac_mode
+
+        zone_control_hvac_mode = self.zone_control_hvac_mode
+
+        # Should be same, because either thermostat or this integration change both.
+        on_off_state = cast(
+            str,
+            self.executor.select_state(
+                OverkizState.CORE_COOLING_ON_OFF
+                if zone_control_hvac_mode == HVACMode.COOL
+                else OverkizState.CORE_HEATING_ON_OFF
+            ),
+        )
+
+        # Device is Stopped, it means the air flux is flowing but its venting door is closed.
+        if on_off_state == OverkizCommandParam.OFF:
+            hvac_mode = HVACMode.OFF
+        else:
+            hvac_mode = zone_control_hvac_mode
+
+        # It helps keep it consistent with the Zone Control, within the interface.
+        if self._attr_hvac_modes != [zone_control_hvac_mode, HVACMode.OFF]:
+            self._attr_hvac_modes = [zone_control_hvac_mode, HVACMode.OFF]
+            self.async_write_ha_state()
```

Comment:
This needs to be removed.

Reply:
If I remove that, it seems the changes I make on _attr_hvac_modes are not instantaneous on the interface...

Reply:
There needs to be a callback method that is called by the thing that initiates the state update. That callback should make all needed changes to attributes and then call `self.async_write_ha_state`.